probe/tls/tls.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package tls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/megaease/easeprobe/global"
+	"github.com/megaease/easeprobe/probe/base"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type TLS struct {
+	base.DefaultOptions `yaml:",inline"`
+	Host                string `yaml:"host"`
+	InsecureSkipVerify  bool   `yaml:"insecure_skip_verify"`
+
+	RootCAPemPath string `yaml:"root_ca_pem_path"`
+	RootCaPem     string `yaml:"root_ca_pem"`
+	rootCAs       *x509.CertPool
```

Comment:
There is a `TLS` struct and `Config()` method in `global/global.go`, do you think we can reuse that?

Reply:
never know that
but the global one introduces extra prop `cert` any `key` which is not only useless but also creating confusion
client cert is not required for tls test

Reply:
the global TLS is originally designed for mTLS, so do you think we can adapt this tls probe for both general TLS and mTLS?

Reply:
Just realized this PR only detects the expiration of certificate, pls ignore the mTLS comments.

And pls add the metric for Prometheus, thanks!